### DISCUSSION TITLE
Array of objects serialization not working on Actions

### DISCUSSION THREAD

**Discussion by VigneshVaidyanathan** [airjp73/rvf#167] [Q&A]
Need help with Array of objects using Remix Validated Form.

I am trying to validate an array of objects using remix-validated-form and it works well during validation when I set the name property in the form as `questions[${index}].placeholder` but when I set it as `questions[${index}][placeholder]` it doesnt validate it properly.

When I use the format `questions[${index}].placeholder`, Remix does not allow to properly serialize it. (Or I might be missing an obvious thing to read using Remix). 

So instead if I use `zod-form-data` to properly serialize it, I have to redo all my schemas that I had for client validation, in server using `zod-form-data`.I also have multiple action types happening in the same action function like `create`, `update`, `create_form` etc, which have different payloads and different schema. So redoing all those again is making it difficult to maintain.

I see the `zod-form-data` is a sugar on top of zod and uses prepreocess and validation logic to parse data into schema types, is there a way to reuse existing zod schema there.

1. Is there a way Remix validated form can validate using the format `questions[${index}][placeholder]` for inputs and arrays. For nested arrays like `questions[${index}][placeholder][0][name]`
2. Is there a way to reuse client validation using Zod inside `zod-form-data` to parse and get the object exactly like the existing zod schema.

**Comment by airjp73** [comment]
Hi!

Would you mind providing a specific example in the form of a code sandbox? The `questions[${index}].placeholder` path looks like it should work so I'd need to see more of what's going on.

I wouldn't expect `questions[${index}][placeholder]` to work because it's not a valid path. Do you mean `questions[${index}]["placeholder"]`?
